firebase-installations/src/main/java/com/google/firebase/installations/CrossProcessLock.java

```diff
@@ -25,67 +26,40 @@
 /** Use file locking to acquire a lock that will also block other processes. */
 class CrossProcessLock {
   private static final String TAG = "CrossProcessLock";
-  private final FileChannel channel;
-  private final FileLock lock;
 
-  private CrossProcessLock(FileChannel channel, FileLock lock) {
-    this.channel = channel;
-    this.lock = lock;
+  interface CrossProcessPersistedInstallation {
+    PersistedInstallationEntry getPersistedInstallations();
   }
 
   /**
-   * Create and return a lock that is exclusive across processes. If another process has the lock
-   * then this call will block until it is released.
+   * Create a lock that is exclusive across processes. If another process has the lock then this
+   * call will block until it is released.
    *
    * @param lockName the lockname is global across all processes of an app
    * @return a CrossProcessLock if success. If the lock failed to acquire (maybe due to disk full or
    *     other unexpected and unsupported permissions) then null will be returned.
    */
-  static CrossProcessLock acquire(Context appContext, String lockName) {
-    FileChannel channel = null;
-    FileLock lock = null;
-    try {
-      File file = new File(appContext.getFilesDir(), lockName);
-      channel = new RandomAccessFile(file, "rw").getChannel();
-      // Use the file channel to create a lock on the file.
-      // This method blocks until it can retrieve the lock.
-      lock = channel.lock();
-      return new CrossProcessLock(channel, lock);
+  static PersistedInstallationEntry whileLocked(
+      Context applicationContext,
+      String lockName,
+      CrossProcessPersistedInstallation persistedInstallation) {
+
+    File file = new File(applicationContext.getFilesDir(), lockName);
```

Comment:
#### Assumption
We always only lock 1 file and it's always the same file per Firebase Installation.

#### Recommendation
If that assumption is correct, I recommend:
* let's initialize the process lock file once when FIS gets initialized (in the constructor)
* let's get rid of the `applicationContext` parameter.